python/prairielearn_test.py

```diff
@@ -1,15 +1,124 @@
 # lol
 import sys
 import os
-sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '../../lib')))
+
+sys.path.insert(
+    0, os.path.abspath(os.path.join(os.path.dirname(__file__), "../../lib"))
+)
 
 import prairielearn as pl  # noqa: E402
-import lxml.html           # noqa: E402
+import lxml.html  # noqa: E402
+import pytest
+from typing import Dict, Optional, Any, Tuple
 
 
 def test_inner_html():
-    e = lxml.html.fragment_fromstring('<div>test</div>')
-    assert pl.inner_html(e) == 'test'
+    e = lxml.html.fragment_fromstring("<div>test</div>")
+    assert pl.inner_html(e) == "test"
+
+    e = lxml.html.fragment_fromstring("<div>test&gt;test</div>")
+    assert pl.inner_html(e) == "test&gt;test"
+
+
+@pytest.mark.parametrize(
+    "question_name, student_answer, weight, expected_grade",
+    [
+        ("base", "a", 1, True),
+        ("base", "a, b", 1, False),
+        ("base", "", 2, False),
+        ("home", "b", 2, True),
+        ("base", "c", 3, True),
+        ("base", "<>", 3, True),
+        ("base", "><", 3, False),
+    ],
+)
+def test_grade_question_parametrized_correct(
+    question_data: pl.QuestionData,
+    question_name: str,
+    student_answer: str,
+    weight: int,
+    expected_grade: bool,
+) -> None:
+
+    question_data["submitted_answers"] = {question_name: student_answer}
```

Comment:
why are the changes in this file part of this PR?

Reply:
oh, I see, you added a new function to `prairielearn.py`.

You should be really careful what type annotations are added to this file, because the typechecking is not running on this file, and the unit tests are being run on a different python version. So we could break production without realizing it. 

Alternatively, enable type checking on this file. But that may require some out-of-scope changes elsewhere in the file. A third option could be to leave the type annotations out of this file for now and then add in a seperate PR, while fixing up the rest of the file for type checking.

Reply:
Added typechecking to this file, although I actually don't think it matters in this case (unless test files are running in production for some reason).

Reply:
oh, right. I guess this comment of mine should have been applied to `prairielearn.py`, which is run in production.

Reply:
Yes. The type signatures from that are taken from our codebase, which is already run in production.